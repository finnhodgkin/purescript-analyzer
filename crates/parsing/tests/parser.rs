macro_rules! parser_tests {
    ($($name:ident => $source:expr),+ $(,)?) => {
        $(
            #[test]
            fn $name() {
                let lexed = lexing::lex($source);
                let tokens = lexing::layout(&lexed);
                let (node, errors) = parsing::parse(&lexed, &tokens);
                insta::assert_debug_snapshot!((node, errors));
            }
        )+
    };
}

parser_tests!(
    comment_pre_post => include_str!("parser/CommentPrePost.purs"),
    lexer_error_inserted => include_str!("parser/LexerErrorInserted.purs"),
    module_export => include_str!("parser/ModuleExport.purs"),
    module_export_class => include_str!("parser/ModuleExportClass.purs"),
    module_export_class_error_end => include_str!("parser/ModuleExportClassErrorEnd.purs"),
    module_export_comma_error => include_str!("parser/ModuleExportCommaError.purs"),
    module_export_data => include_str!("parser/ModuleExportData.purs"),
    module_export_data_comma_error => include_str!("parser/ModuleExportDataCommaError.purs"),
    module_export_data_item_after_double_period => include_str!("parser/ModuleExportDataItemAfterDoublePeriod.purs"),
    module_export_data_item_before_double_period => include_str!("parser/ModuleExportDataItemBeforeDoublePeriod.purs"),
    module_export_data_item_invalid => include_str!("parser/ModuleExportDataItemInvalid.purs"),
    module_export_empty => include_str!("parser/ModuleExportEmpty.purs"),
    module_export_invalid => include_str!("parser/ModuleExportInvalid.purs"),
    module_export_type_operator => include_str!("parser/ModuleExportTypeOperator.purs"),
    module_export_type_operator_error => include_str!("parser/ModuleExportTypeOperatorError.purs"),
    module_export_type_operator_error_end => include_str!("parser/ModuleExportTypeOperatorErrorEnd.purs"),
    module_header => include_str!("parser/ModuleHeader.purs"),
    module_header_prefixed => include_str!("parser/ModuleHeaderPrefixed.purs"),
);
