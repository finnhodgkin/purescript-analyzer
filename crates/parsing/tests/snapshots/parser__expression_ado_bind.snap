---
source: crates/parsing/tests/parser.rs
expression: "(node, errors)"
snapshot_kind: text
---
(
    Module@0..71
      ModuleHeader@0..32
        MODULE@0..6 "module"
        Comment@6..7
          WHITESPACE@6..7 " "
        ModuleName@7..25
          UPPER@7..24 "ExpressionAdoBind"
          Comment@24..25
            WHITESPACE@24..25 " "
        ModuleExportList@25..25
        WHERE@25..30 "where"
        Comment@30..32
          WHITESPACE@30..32 "\n\n"
      LAYOUT_START@32..32 ""
      ModuleImports@32..32
      ModuleStatements@32..71
        ValueEquation@32..71
          LOWER@32..36 "main"
          Comment@36..37
            WHITESPACE@36..37 " "
          BindersList@37..37
          EquationUnguarded@37..71
            EQUAL@37..38 "="
            Comment@38..39
              WHITESPACE@38..39 " "
            EquationWhere@39..71
              ExpressionAdo@39..71
                ADO@39..42 "ado"
                Comment@42..45
                  WHITESPACE@42..45 "\n  "
                DoStatements@45..63
                  LAYOUT_START@45..45 ""
                  DoStatementBind@45..63
                    BinderVariable@45..50
                      LOWER@45..49 "life"
                      Comment@49..50
                        WHITESPACE@49..50 " "
                    LEFT_ARROW@50..52 "<-"
                    Comment@52..53
                      WHITESPACE@52..53 " "
                    ExpressionApplicationChain@53..63
                      ExpressionVariable@53..58
                        LOWER@53..57 "pure"
                        Comment@57..58
                          WHITESPACE@57..58 " "
                      ExpressionTermArgument@58..63
                        ExpressionInteger@58..63
                          INTEGER@58..60 "42"
                          Comment@60..63
                            WHITESPACE@60..63 "\n  "
                  LAYOUT_END@63..63 ""
                IN@63..65 "in"
                Comment@65..66
                  WHITESPACE@65..66 " "
                ExpressionVariable@66..71
                  LOWER@66..70 "life"
                  Comment@70..71
                    WHITESPACE@70..71 "\n"
      LAYOUT_END@71..71 ""
      END_OF_FILE@71..71 ""
    ,
    [],
)
